cmd/git-go-patch/extract.go

```diff
@@ -43,145 +56,138 @@ Each command starts with "` + commandPrefix + `", then:
   distinct groups of patches. Making the first patch in each group start at a consistent number can
   help to avoid unnecessary filename conflicts when porting changes between branches.
 ` + repoRootSearchDescription
+}
+func (e extractCmd) Handle(p subcmd.ParseFunc) error {
```

Comment:
One thought I had reading through this is there is a lot of IO happening and when it fails we resort to 

```go
return err
```

I'm considering how that will look when we have an error in an AzDO run. It will result in a failure that has an IO message in the logs but it seems like it will be really hard to track that back down to what failed. Basically if we have a "Access Denied" error in the logs how will we tell which of the various IO paths in this code was responsible? 

Is it worth adding logs on the failure cases so we have a trail to look at?

Reply:
In a lot of cases I think we can pretty easily figure out where an error happened by looking at the last positive/progress log line. But, I try to always remember to wrap errors when I think there might be some ambiguity, like `return fmt.Errorf("unable to chop the carrot: %w", err)` when `err` could be `knife is dull` but the func is making an entire bowl of soup.

In my experience, IO errors always mention the filename, so there are a decent number of situations where even without wrapping, the error would still be obvious.

This seems pretty subjective to me, and wrapping too much is definitely better than not wrapping enough, so I'd be happy to put more in. (Just maybe not in `git-go-patch` because this isn't new code, only unindented.)